src/main/java/software/amazon/encryption/s3/materials/RsaKeyring.java

```diff
@@ -97,6 +97,11 @@ public String keyProviderInfo() {
             return KEY_PROVIDER_INFO;
         }
 
+        @Override
+        public EncryptionMaterials generateDataKey(EncryptionMaterials materials) {
+            return defaultGenerateDataKey(materials);
```

Comment:
editorial: I wish we could avoid having to do this, but after staring at it awhile and trying a few things, I don't think it's possible. The problem is that the S3Keyring contains the reference to the `_dataKeyGenerator` which MUST provide the randomness, and the `generateDataKey()` method is defined in `GenerateDataKeyStrategy`, which is implemented as an inner class inside of the various keyrings. So, annoyingly, you can access the `_dataKeyGenerator` from within the inner class (`_rsaOaepStrategy` in this case) but you can't from the `GenerateDataKeyStrategy` interface, which is where the `default` method would need to go. It might be possible by refactoring to make `DataKeyGenerator` a member of `DataKeyStrategy` but that's more trouble than it's worth. 

I can't think of a better solution than what you have, so let's go with that :)